.changeset/purple-hats-roll.md

```diff
@@ -0,0 +1,18 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+---
+---
+"section": "tree"
+---
+RestrictiveReadonlyRecord is deprecated
+
+`RestrictiveReadonlyRecord` was an attempt to implement a version of the built in `Record<TKey, TValue>` type that would prohibit (instead of leaving unrestricted like Record does) values under keys that are do not extend `TKey`.
```

Comment:
```suggestion
`RestrictiveReadonlyRecord` was an attempt to implement a version of the built-in `Record<TKey, TValue>` type that would prohibit (instead of leaving unrestricted like Record does) values under keys that do not extend `TKey`.
```
```suggestion
`RestrictiveReadonlyRecord` was an attempt to implement a version of the built in `Record<TKey, TValue>` type that would prohibit (instead of leaving unrestricted like Record does) values under keys that are do not extend `TKey`.
```